src/Lucene.Net/Index/TermsHashPerField.cs

```diff
@@ -355,168 +355,176 @@ public void  Add(int textStart)
-					RehashPostings(2 * postingsHashSize);
-				
-				// Init stream slices
-				if (numPostingInt + intPool.intUpto > DocumentsWriter.INT_BLOCK_SIZE)
-					intPool.NextBuffer();
-				
-				if (DocumentsWriter.BYTE_BLOCK_SIZE - bytePool.byteUpto < numPostingInt * ByteBlockPool.FIRST_LEVEL_SIZE)
-					bytePool.NextBuffer();
-				
-				intUptos = intPool.buffer;
-				intUptoStart = intPool.intUpto;
-				intPool.intUpto += streamCount;
-				
-				p.intStart = intUptoStart + intPool.intOffset;
-				
-				for (int i = 0; i < streamCount; i++)
-				{
-					int upto = bytePool.NewSlice(ByteBlockPool.FIRST_LEVEL_SIZE);
-					intUptos[intUptoStart + i] = upto + bytePool.byteOffset;
-				}
-				p.byteStart = intUptos[intUptoStart];
-				
-				consumer.NewTerm(p);
-			}
-			else
-			{
-				intUptos = intPool.buffers[p.intStart >> DocumentsWriter.INT_BLOCK_SHIFT];
-				intUptoStart = p.intStart & DocumentsWriter.INT_BLOCK_MASK;
-				consumer.AddTerm(p);
-			}
-			
-			if (doNextCall)
-				nextPerField.Add(p.textStart);
-		}
+                    postingsHash[hashPos] = p;
+                    numPostings++;
+
+                    if (numPostings == postingsHashHalfSize)
+                        RehashPostings(2 * postingsHashSize);
+
+                    // Init stream slices
+                    if (numPostingInt + intPool.intUpto > DocumentsWriter.INT_BLOCK_SIZE)
+                        intPool.NextBuffer();
+
+                    if (DocumentsWriter.BYTE_BLOCK_SIZE - bytePool.byteUpto < numPostingInt * ByteBlockPool.FIRST_LEVEL_SIZE)
+                        bytePool.NextBuffer();
+
+                    intUptos = intPool.buffer;
+                    intUptoStart = intPool.intUpto;
+                    intPool.intUpto += streamCount;
+
+                    p.intStart = intUptoStart + intPool.intOffset;
+
+                    for (int i = 0; i < streamCount; i++)
+                    {
+                        int upto = bytePool.NewSlice(ByteBlockPool.FIRST_LEVEL_SIZE);
+                        intUptos[intUptoStart + i] = upto + bytePool.byteOffset;
+                    }
+                    p.byteStart = intUptos[intUptoStart];
+
+                    consumer.NewTerm(p);
+                }
+                else
+                {
+                    intUptos = intPool.buffers[p.intStart >> DocumentsWriter.INT_BLOCK_SHIFT];
+                    intUptoStart = p.intStart & DocumentsWriter.INT_BLOCK_MASK;
+                    consumer.AddTerm(p);
+                }
+
+                if (doNextCall)
+                    nextPerField.Add(p.textStart);
+            }
+            catch (Exception e)
+            {
+                var field = docState.doc.GetField("id()");
+                throw new InvalidOperationException($"Failed to add term for document field '{fieldInfo.name}', docId: '{field?.fieldsData}', term: '{termAtt.Term}', tokenText: '{new string(tokenText)}', tokenTextLength: '{tokenTextLen}'.", e);
```

Comment:
hard for me to follow changes here, except of wrapping everything in try/catch are there any other changes?
I'm a bit worried about wrapping each exception in 'InvalidOperationException' - is there a specific exception we want to catch? Are there any expected exception that we shouldn't catch?

Reply:
A refactoring was carried out, in which:
1. Error handling moved to caller level.
2. Only the error that prompted us to make these changes is processed (`IndexOutOfRangeException`)

The information content of the error message is preserved.
An example of a previous implementation error:
`Failed to add term for document field 'Name', docId: 'contract/1-A', term: 'quotes/98ed381e-116a-46c4-b25a-6c7d4e89c239'`

Now the message looks like this:
`Failed to add term for document field 'Name', docId: 'contract/1-A', field attributes: (startOffset=0,endOffset=43,term=quotes/98ed381e-116a-46c4-b25a-6c7d4e89c239 ,positionIncrement=1)`